src/main/java/fi/aalto/cs/apluscourses/model/SubmissionStatusUpdater.java

```diff
@@ -109,9 +109,13 @@ private void run() {
         try {
           submissionResult =
               dataSource.getSubmissionResult(submissionUrl, exercise, authentication);
-          if (submissionResult.getStatus() != SubmissionResult.Status.UNKNOWN) {
+          if (SubmissionResult.Status.WAITING.equals(submissionResult.getStatus())) {
+            PluginSettings.getInstance().getMainViewModel(project).setSubmittedForGrading(exercise);
+            PluginSettings.getInstance().updateMainViewModel(project);
```

Comment:
An update includes the following:

- Fetch the course configuration JSON and parse it again.
- Fetch a list of all exercises from the A+ API
- Fetch a list of points for the student.

In this case that's a bit too much, because this gets run repeatedly in increments (0 s, 10 s, 25 s, 45 s, etc). I assume this is here so that the icon gets updated? If that's the case, I have an idea for how to avoid the update, I'll open a PR against this branch